components/org.wso2.carbon.stream.processor.core/src/main/java/org/wso2/carbon/stream/processor/core/ha/HACoordinationSourceHandler.java

```diff
@@ -154,85 +154,95 @@ public String getElementId() {
     }
 
     private void sendEventsToPassiveNode(Event event, String[] transportSyncProperties) {
-        EventSyncConnection eventSyncConnection = getTCPNettyClient();
-        ByteBuffer messageBuffer = null;
-        if (eventSyncConnection != null) {
-            QueuedEvent queuedEvent = new QueuedEvent(siddhiAppName, sourceHandlerElementId, sequenceIDGenerator
-                    .incrementAndGet(), event, transportSyncProperties);
-            try {
-                messageBuffer = BinaryEventConverter.convertToBinaryMessage(new QueuedEvent[]{queuedEvent});
-            } catch (IOException e) {
-                log.error("Error in converting events to binary message.Hence not sending message to the passive node");
-                return;
-            }
-            if (messageBuffer != null) {
+        EventSyncConnection eventSyncConnection = null;
+        try {
+            eventSyncConnection = getTCPNettyClient();
+            ByteBuffer messageBuffer = null;
+            if (eventSyncConnection != null) {
+                QueuedEvent queuedEvent = new QueuedEvent(siddhiAppName, sourceHandlerElementId, sequenceIDGenerator
+                        .incrementAndGet(), event, transportSyncProperties);
                 try {
-                    eventSyncConnection.send(HAConstants.CHANNEL_ID_MESSAGE, messageBuffer.array());
-                    if (throughputTracker != null && StreamProcessorDataHolder.isStatisticsEnabled()) {
-                        throughputTracker.eventIn();
+                    messageBuffer = BinaryEventConverter.convertToBinaryMessage(new QueuedEvent[]{queuedEvent});
+                } catch (IOException e) {
+                    log.error("Error in converting events to binary message.Hence not sending message to the passive node");
```

Comment:
Any reason on, why can't we add the error trace here? This is a genuine error which we should print the trace as well. Isn't it?

Reply:
added the trace